src/migrateDiagram.js

```diff
@@ -29,16 +29,25 @@ const DMN13URI = '"https://www.omg.org/spec/DMN/20191111/MODEL/"';
  * @returns {Promise<string>}
  */
 export function migrateDiagram(xml) {
+  return new Promise((resolve, reject) => {
+    if (!isString(xml)) {
+      reject(new Error(`expected DMN diagram, found ${ xml }`));
+    }
 
-  if (hasNamespace(DMN12URI, xml)) {
-    return migrateFrom12To13(xml);
-  } else if (hasNamespace(DMN11URI, xml)) {
-    return migrateFrom11To13(xml);
-  } else if (hasNamespace(DMN13URI, xml)) {
-    return Promise.resolve(xml);
-  }
+    if (hasNamespace(DMN11URI, xml)) {
+      try {
+        resolve(migrateFrom11To13(xml));
+      } catch (err) {
+        reject(err);
+      }
+    } else if (hasNamespace(DMN12URI, xml)) {
+      resolve(migrateFrom12To13(xml));
+    } else if (hasNamespace(DMN13URI, xml)) {
+      resolve(xml);
+    }
 
-  throw new Error('unknown namespace');
+    reject(new Error(`expected DMN diagram, found ${ xml }`));

```

Comment:
We always end up here due to missing return statements.

Reply:
💩